users/danielo515/danielo515.c

```diff
@@ -0,0 +1,373 @@
+#include "danielo515.h"
+
+bool onMac = true;
+// Send control or GUI depending if we are on windows or mac
+bool CMD(uint16_t kc) {
+  if(onMac){ tap_code16(LGUI(kc)); } else { tap_code16(LCTL(kc)); }
+  return false;
+}
+
+//**************** Handle keys function *********************//
+bool altPressed = false;
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record)
+{
+  bool pressed = record->event.pressed;
+  if(pressed){
+    refresh_incremental_macros(keycode);
+    if(process_incremental_macro(keycode)){
+      return false;
+    }
+    if(is_macro(keycode)){
+      return handle_macro(keycode);
+    }
+   switch (keycode) {
+        case MAC_TGL:
+        onMac = !onMac;
+        onMac ? SEND_STRING("On mac") : SEND_STRING("Not on MAC");
+        return false;
+      }
+  }
+
+  switch (keycode)
+  {
+    case QWERTY:
+        if (record->event.pressed) {
+        #ifdef AUDIO_ENABLE
+        PLAY_SONG(tone_qwerty);
+        #endif
+        layer_on(_QWERTY);
+        }
+        return false;
+    case LOWER:
+        if (record->event.pressed) {
+        layer_on(_LOWER);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+        } else {
+        layer_off(_LOWER);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+        }
+        return false;
+    case RAISE:
+        if (record->event.pressed) {
+        layer_on(_RAISE);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+        } else {
+        layer_off(_RAISE);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+        }
+        return false;
+    case ADJUST:
+    if (record->event.pressed) {
+        layer_on(_ADJUST);
+        } else {
+        layer_off(_ADJUST);
+        }
+        return false;
+ // == Macros START ===
+  case ARROW:
+    if (record->event.pressed) SEND_STRING("->");
+    return false;
+  case F_ARROW:
+    if (record->event.pressed) SEND_STRING("=>");
+    return false;
+  case GREP:
+    if (record->event.pressed) SEND_STRING(" | grep "); return false;
+ // == Macros END ===
+ // == Multi Os START ===
+  case KC_HOME:// make the home behave the same on OSX
+    if (record->event.pressed && onMac) {
+        SEND_STRING(SS_LCTRL("a"));
+        return false;
+    }
+  case KC_END:// make the end behave the same on OSX
+    if (record->event.pressed && onMac) {
+        SEND_STRING(SS_LCTRL("e"));
```

Comment:
You can save some bytes here by using `tap_code16()`s, eg:
```c
tap_code16(C(KC_E));
```

Reply:
2178 -> 2180 bytes free. Every byte counts!
Thank you, any other place where you think I can apply the same optimization?